File: src/main/java/com/busted_moments/client/features/war/WarHornFeature.java
package com.busted_moments.client.features.war;

import com.busted_moments.client.config.providers.sound.SoundProvider;
import com.busted_moments.client.models.war.timer.events.TimerStartEvent;
import com.busted_moments.client.util.SoundUtil;
import com.busted_moments.client.util.Sounds;
import com.busted_moments.core.Default;
import com.busted_moments.core.Feature;
import com.busted_moments.core.State;
import com.busted_moments.core.config.Config;
import net.minecraft.sounds.SoundEvent;
import net.minecraftforge.eventbus.api.SubscribeEvent;

@Default(State.ENABLED)
@Config.Category("War")
@Feature.Definition(name = "War Horn", description = "Plays a sound when a war starts")
public class WarHornFeature extends Feature {
   @Dropdown(title = "Selected Sound", options = SoundProvider.class)
   private SoundEvent selected = Sounds.WAR_HORN;

   @Value("Sound volume")
   @Tooltip("The volume of the sound")
   private static Float volume = 1.0F;

   @Value("Pitch")
   @Tooltip("The pitch of the sound")
   private static Float pitch = 1.0F;

   @SubscribeEvent
   public void onTimerStart(TimerStartEvent event) {
      if (event.getTimer().isConfident()) SoundUtil.playAmbient(selected, pitch, volume);
   }
}

File: src/main/kotlin/com/busted_moments/client/features/war/WarHornFeature.kt
package com.busted_moments.client.features.war

import com.busted_moments.client.framework.config.annotations.Category
import com.busted_moments.client.framework.config.entries.dropdown.Dropdown
import com.busted_moments.client.framework.config.entries.value.Value
import com.busted_moments.client.framework.events.Subscribe
import com.busted_moments.client.framework.features.Feature
import com.busted_moments.client.framework.features.Replaces
import com.busted_moments.client.framework.sounds.Sounds
import com.busted_moments.client.framework.sounds.Sounds.play
import com.busted_moments.client.models.territories.timers.events.TimerEvent
import com.wynntils.features.embellishments.WarHornFeature
import net.minecraft.sounds.SoundEvent
import net.minecraft.sounds.SoundSource

@Category("War")
@Replaces(WarHornFeature::class)
object WarHornFeature : Feature() {
    @Dropdown("Selected Sound")
    private var sound: SoundEvent = Sounds.WAR_HORN

    @Value("Source")
    private var source: SoundSource = SoundSource.AMBIENT

    @Value("Sound volume")
    private var volume: Float = 1.0f

    @Value("Speed")
    private var speed: Float = 1.0f

    @Subscribe
    private fun TimerEvent.Enqueued.on() {
        if (source == TimerEvent.Source.CHAT)
            sound.play(
                source = this@WarHornFeature.source,
                volume = volume,
                speed = speed
            )
    }
}